packages/amplify-graphql-auth-transformer/src/resolvers/helpers.ts

```diff
@@ -168,19 +168,23 @@ export const iamExpression = (
 /**
  * Creates iam admin role check helper
  */
-export const iamAdminRoleCheckExpression = (adminRoles: Array<string>, fieldName?: string): Expression => compoundExpression([
-  set(ref('adminRoles'), raw(JSON.stringify(adminRoles))),
-  forEach(/* for */ ref('adminRole'), /* in */ ref('adminRoles'), [
-    iff(
-      and([
-        methodCall(ref('ctx.identity.userArn.contains'), ref('adminRole')),
-        notEquals(ref('ctx.identity.userArn'), ref('ctx.stash.authRole')),
-        notEquals(ref('ctx.identity.userArn'), ref('ctx.stash.unauthRole')),
-      ]),
-      fieldName ? raw(`#return($context.source.${fieldName})`) : raw('#return($util.toJson({}))'),
-    ),
-  ]),
-]);
+export const iamAdminRoleCheckExpression = (adminRoles: Array<string>, fieldName?: string, adminCheckExpression?: Expression): Expression => {
+  const returnStatement = fieldName ? raw(`#return($context.source.${fieldName})`) : raw('#return($util.toJson({}))');
+  const fullReturnExpression = adminCheckExpression ? compoundExpression([ adminCheckExpression, returnStatement ]) : returnStatement;
```

Comment:
Is there a chance that `fullReturnExpression` is `undefined`? (if it is used in other code snippet)

Reply:
No. The `returnStatement` always exists, so the `fullReturnExpression` cannot be undefined.